deploy/charts/preflight/templates/agent_installation.yaml

```diff
@@ -0,0 +1,214 @@
+  - mutatingwebhookconfigurations
+  verbs: ["get", "list"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-secret-reader
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-secret-reader
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-cert-manager-reader
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-cert-manager-reader
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-googlecas-reader
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-googlecas-reader
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-get-webhooks
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-get-webhooks
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-node-reader
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  kind: ClusterRole
+  name: jetstack-secure-agent-node-reader
+  apiGroup: rbac.authorization.k8s.io
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: jetstack-secure-agent-cluster-viewer
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: view
+subjects:
+- kind: ServiceAccount
+  name: agent
+  namespace: {{ .Release.Namespace }}
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: agent
+  labels:
+    {{- include "preflight.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app.kubernetes.io/name: agent
+      {{- include "preflight.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "preflight.selectorLabels" . | nindent 8 }}
+    spec:
+      serviceAccountName: agent
```

Comment:
I think the service account name needs to be configurable so that we can swap it out in the Marketplace release. I guess we can take the example [from cert-manager](https://github.com/jetstack/cert-manager/blob/master/deploy/charts/cert-manager/templates/deployment.yaml#L53):

```
      serviceAccountName: {{ index .Values "cert-manager" "serviceAccountName" }}
```

Note the `index .Values`: this is due to the fact that Helm does not like JSON fields that contain a dash, so instead of doing

```
{{ .Values.cert-manager.serviceAccountName }}
```

we instead have to do:

```
{{ index .Values "cert-manager" "serviceAccountName" }}
```

Reply:
There's a `serviceAccountName` defined in the _helpers.tpl file, so I used that. It's derived from the values.yaml value or the release name.